engine/graphics/src/graphics.h

```diff
@@ -216,33 +219,34 @@ namespace dmGraphics
         , m_VWrap(TEXTURE_WRAP_CLAMP_TO_EDGE)
         , m_Data(0x0)
         , m_DataSize(0)
-        , m_MipMap(0)
+        , m_X(0)
+        , m_Y(0)
+        , m_Z(0)
         , m_Width(0)
         , m_Height(0)
         , m_Depth(0)
+        , m_MipMap(0)
         , m_SubUpdate(false)
-        , m_X(0)
-        , m_Y(0)
-        , m_Z(0)
         {}
 
         TextureFormat m_Format;
         TextureFilter m_MinFilter;
         TextureFilter m_MagFilter;
-        TextureWrap m_UWrap;
-        TextureWrap m_VWrap;
-        const void* m_Data;
-        uint32_t m_DataSize;
-        uint16_t m_MipMap;
-        uint16_t m_Width;
-        uint16_t m_Height;
-        uint16_t m_Depth; // For array texture, this is slice count
+        TextureWrap   m_UWrap;
+        TextureWrap   m_VWrap;
+        const void*   m_Data;
+        uint32_t      m_DataSize;
 
         // For sub texture updates
-        bool m_SubUpdate;
         uint32_t m_X;
         uint32_t m_Y;
         uint32_t m_Z; // For array texture, this is the slice to set
+
+        uint16_t m_Width;
+        uint16_t m_Height;
+        uint16_t m_Depth; // For array texture, this is slice count
+        uint8_t  m_MipMap    : 7;
+        uint8_t  m_SubUpdate : 1;
```

Comment:
What's the estimated upper limit of the depth? 255?
Then it could be:
```
uint16_t m_Depth : 8;
uint16_t m_MipMap : 7;
uint16_t m_SubUpdate : 1;
```

Reply:
The upper depth limit is dictated by the graphics card vendors for 3D/array textures, I don't think we have an absolute max for depth slices in our engine. We loosely support 16k textures so the depth should also adhere to that limit